### ISSUE TITLE
NO-MATCHING of current module

### ISSUE THREAD

**Issue by ZhuYun97** [KaihuaTang/Scene-Graph-Benchmark.pytorch#28]
I download the pre-trained model [predcls](https://onedrive.live.com/embed?cid=22376FFAD72C4B64&resid=22376FFAD72C4B64%21781937&authkey=AOzowl5-07RzJz4), and run as below:
`CUDA_VISIBLE_DEVICES=0 python -m torch.distributed.launch --master_port 10027 --nproc_per_node=1 tools/relation_test_net.py --config-file "configs/e2e_relation_X_101_32_8_FPN_1x.yaml" MODEL.ROI_RELATION_HEAD.USE_GT_BOX True MODEL.ROI_RELATION_HEAD.USE_GT_OBJECT_LABEL True MODEL.ROI_RELATION_HEAD.PREDICTOR MotifPredictor TEST.IMS_PER_BATCH 1 DTYPE "float16" GLOVE_DIR /home/mist/glove MODEL.PRETRAINED_DETECTOR_CKPT /home/mist/checkpoints/motif-precls/model_0030000.pth OUTPUT_DIR /home/mist/checkpoints/motif-precls MODEL.WEIGHT /home/mist/checkpoints/motif-precls/model_0030000.pth
`
**But it seems the checkpoint lacks some modules. And the results are also wired. mAP detection is around 100%, and other metrics are all lower than you reported. What's wrong with it?**

```
2020-05-07 05:34:12,114 maskrcnn_benchmark.utils.model_serialization INFO: NO-MATCHING of current module: roi_heads.relation.predictor.post_cat.bias of shape (4096,)
2020-05-07 05:34:12,114 maskrcnn_benchmark.utils.model_serialization INFO: NO-MATCHING of current module: roi_heads.relation.predictor.post_cat.weight of shape (4096, 1024)
2020-05-07 05:34:12,115 maskrcnn_benchmark.utils.model_serialization INFO: NO-MATCHING of current module: roi_heads.relation.predictor.rel_compress.bias of shape (51,)
2020-05-07 05:34:12,115 maskrcnn_benchmark.utils.model_serialization INFO: NO-MATCHING of current module: roi_heads.relation.predictor.rel_compress.weight of shape (51, 4096)
2020-05-07 05:34:17,258 maskrcnn_benchmark.inference INFO: Start evaluation on VG_stanford_filtered_with_attribute_test dataset(26446 images).
```

The result shows below
```
DONE (t=32.50s).
 Average Precision  (AP) @[ IoU=0.50:0.95 | area=   all | maxDets=100 ] = 0.999
 Average Precision  (AP) @[ IoU=0.50      | area=   all | maxDets=100 ] = 0.999
 Average Precision  (AP) @[ IoU=0.75      | area=   all | maxDets=100 ] = 0.999
 Average Precision  (AP) @[ IoU=0.50:0.95 | area= small | maxDets=100 ] = 0.999
 Average Precision  (AP) @[ IoU=0.50:0.95 | area=medium | maxDets=100 ] = 1.000
 Average Precision  (AP) @[ IoU=0.50:0.95 | area= large | maxDets=100 ] = 1.000
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=  1 ] = 0.659
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets= 10 ] = 0.995
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=100 ] = 1.000
 Average Recall     (AR) @[ IoU=0.50:0.95 | area= small | maxDets=100 ] = 1.000
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=medium | maxDets=100 ] = 1.000
 Average Recall     (AR) @[ IoU=0.50:0.95 | area= large | maxDets=100 ] = 1.000

2020-05-07 07:08:52,592 maskrcnn_benchmark.inference INFO: 
====================================================================================================
Detection evaluation mAp=0.9995
====================================================================================================
SGG eval:   R @ 20: 0.4845;   R @ 50: 0.5942;   R @ 100: 0.6388;  for mode=predcls, type=Recall(Main).
SGG eval: ngR @ 20: 0.5314; ngR @ 50: 0.7008; ngR @ 100: 0.8024;  for mode=predcls, type=No Graph Constraint Recall(Main).
SGG eval:  zR @ 20: 0.0002;  zR @ 50: 0.0013;  zR @ 100: 0.0022;  for mode=predcls, type=Zero Shot Recall.
SGG eval:  mR @ 20: 0.0829;  mR @ 50: 0.1276;  mR @ 100: 0.1549;  for mode=predcls, type=Mean Recall.
(above:0.0854) (across:0.0000) (against:0.0000) (along:0.0092) (and:0.0237) (at:0.1920) (attached to:0.0024) (behind:0.4910) (belonging to:0.0000) (between:0.0069) (carrying:0.2284) (covered in:0.1619) (covering:0.0129) (eating:0.4193) (flying in:0.0000) (for:0.1236) (from:0.0282) (growing on:0.0000) (hanging from:0.0372) (has:0.7925) (holding:0.5807) (in:0.3445) (in front of:0.0680) (laying on:0.0045) (looking at:0.0311) (lying on:0.0000) (made of:0.0625) (mounted on:0.0000) (near:0.2985) (of:0.6079) (on:0.7632) (on back of:0.0000) (over:0.0823) (painted on:0.0000) (parked on:0.0000) (part of:0.0000) (playing:0.0000) (riding:0.4037) (says:0.0000) (sitting on:0.2931) (standing on:0.0066) (to:0.0000) (under:0.2259) (using:0.2667) (walking in:0.0000) (walking on:0.0134) (watching:0.0726) (wearing:0.9630) (wears:0.0008) (with:0.0394) 
SGG eval:   A @ 20: 0.6770;   A @ 50: 0.6795;   A @ 100: 0.6795;  for mode=predcls, type=TopK Accuracy.
====================================================================================================
```

**Comment by KaihuaTang** [comment]
mAp=0.9995 because you are running **PredCls** protocol, which takes ground-truth bounding boxes and object labels as input.  Only **SGDet** directly predicts relationships from the raw image, i.e., USE_GT_BOX  False, USE_GT_OBJECT_LABEL False.  There are two modules fail to be loaded as you said, I think that's why the PredCls results are lower than the reported.  
The reason why these two modules are missing, because the checkpoint I uploaded is not **MotifPredictor** but **CausalAnalysisPredictor** with CAUSAL.CONTEXT_LAYER motifs (the variable names are different so does the fusion function.)

**Comment by ZhuYun97** [comment]
If I want to use MotifPredictor. Is it possible to train the model(build on your pre-trained model) to reach the accuracy
